eth/api_tracer.go

```diff
@@ -352,23 +353,29 @@ func (api *PrivateDebugAPI) traceChain(ctx context.Context, start, end *types.Bl
 
 // TraceBlockByNumber returns the structured logs created during the execution of
 // EVM and returns them as a JSON object.
-func (api *PrivateDebugAPI) TraceBlockByNumber(ctx context.Context, number rpc.BlockNumber, config *TraceConfig) ([]*txTraceResult, error) {
+func traceBlockByNumber(ctx context.Context, eth *Ethereum, number rpc.BlockNumber, config *TraceConfig) ([]*txTraceResult, error) {
```

Comment:
What's the reasoning behind pulling the receiver out and using the `eth *Ethereum` arg instead? (Instead of just making the method private?) Not against it, just curious.

Reply:
Just for using it in both **PrivateDebugAPI** and **PrivateTraceAPI** [here](https://github.com/etclabscore/core-geth/pull/211/files#diff-e98ceee1b6a777b2ee45a6f6527b9d51da990c691cbb19997c8c104ca9dd4725R136). I would be interested if you have a better idea on doing this, as I don't like it much. I did the same for more functions.